### DISCUSSION TITLE
Can you create a dump with LOGS table, but without its content?

### DISCUSSION THREAD

**Discussion by kasteckis** [spatie/laravel-db-snapshots#137] [Q&A] [answered]
.

**Comment by rawilk** [comment, marked as answer]
I had a similar use case and I solved it adding extra dump options to the config for my database connection. In my use case, I actually have several tables that I want to include the schema for, but without the data. Here is a simplified version of my custom command I wrote to do this particular snapshot:

```php
public function handle()
{
    $productionConnectionName = config('database.default');
    config()->set('database.connections.{$productionConnectionName}.dump", [
        'addExtraOption' => $this->buildDumpOptions(),
    ]);

    $this->call(Create::class, [
        '--connection' => $productionConnectionName,
    ]);
}

protected function buildDumpOptions(): string
{
    $options = [
        '--no-owner',

        ...array_map(
            fn (string $table) => "--exclude-table-data={$table}",
            config('site.staging.exclude_tables'),
        ),
    ];

    return implode(' ', $options);
}
```

**Note:** My application is using postgres, so the `pg_dump` option I needed was `--exclude-table-data`, but I'm sure MySQL or SQLite have similar options.
